PyChaste conda VTK error for Python 3.9 - 3.11

Fix VTK related error on PyChaste conda CI (https://github.com/Chaste/Chaste/actions/runs/17677751175/)

```
-- Configuring done (110.3s)
CMake Error at /home/runner/miniconda3/envs/test/lib/cmake/vtk-9.2/VTK-targets.cmake:493 (set_target_properties):
  The link interface of target "VTK::opengl" contains:

    OSMesa::OSMesa

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /home/runner/miniconda3/envs/test/lib/cmake/vtk-9.2/vtk-config.cmake:138 (include)
  CMakeLists.txt:393 (find_package)
```

These tests run OK for Python 3.12 and 3.13, but config fails for Python 3.9 - 3.11

--- a/.github/workflows/pychaste-conda-tests.yml
+++ b/.github/workflows/pychaste-conda-tests.yml
@@ -20,7 +20,7 @@ jobs:
       matrix:
         python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
 
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
 
     env:
       CHASTE_TEST_OUTPUT: ${{ github.workspace }}/testoutput
@@ -41,7 +41,7 @@ jobs:
       - name: Install system dependencies
         run: |
           sudo apt-get update
-          sudo apt-get install -y castxml clang cmake libx11-dev xvfb
+          sudo apt-get install -y castxml clang cmake libosmesa6-dev libpthread-stubs0-dev libx11-dev pkg-config xvfb
 
       - name: Setup Miniconda Python ${{ matrix.python-version }}
         uses: conda-incubator/setup-miniconda@v3

